torch_frame/nn/models/bcauss.py

```diff
@@ -0,0 +1,138 @@
+            ReLU(),
+            Linear(hidden_channels, out_channels)
+        ])
+        self.reset_parameters()
+
+    def reset_parameters(self) -> None:
+        for block in self.model:
+            if isinstance(block, Linear):
+                block.reset_parameters()
+
+    def forward(self, x: Tensor) -> Tensor:
+        r"""Transforming :obj:`x` into output predictions.
+
+        Args:
+            x (Tensor): Input column-wise tensor of shape
+                [batch_size, num_cols, in_channels]
+
+        Returns:
+            Tensor: [batch_size, out_channels].
+        """
+        return self.model(x)
+
+
+class BalanceScoreEstimator(Module):
+    def __init__(self, channels, out_channels):
+        super().__init__()
+        self.model = Linear(channels, out_channels)
+
+    def reset_parameters(self):
+        self.model.reset_parameters()
+
+    def forward(self, x: Tensor):
+        return F.sigmoid(self.model(x))
+
+
+class BCAUSS(Module):
+    def __init__(
+        self,
+        channels: int,
+        hidden_channels: int,
+        decoder_hidden_channels: int,
+        out_channels: int,
+        col_stats: dict[str, dict[StatType, Any]] | None,
+        col_names_dict: dict[torch_frame.stype, list[str]] | None,
+        epsilon: float = 0.5,
+    ):
+
+        super().__init__()
+        if col_stats is not None and col_names_dict is not None:
+            numerical_stats_list = [
+                col_stats[col_name]
+                for col_name in col_names_dict[torch_frame.numerical]
+            ]
+            mean = torch.tensor(
+                [stats[StatType.MEAN] for stats in numerical_stats_list])
+            self.register_buffer("mean", mean)
+            std = (torch.tensor(
+                [stats[StatType.STD]
+                 for stats in numerical_stats_list]) + 1e-6)
+            self.register_buffer("std", std)
+        self.representation_learner = MLPBlock(channels, hidden_channels,
+                                               hidden_channels)
+        self.balance_score_learner = BalanceScoreEstimator(
+            hidden_channels, out_channels)
+        # decoder for treatment group
+        self.treatment_decoder = MLPBlock(hidden_channels,
+                                          decoder_hidden_channels,
+                                          out_channels)
+        # decoder for control group
+        self.control_decoder = MLPBlock(hidden_channels,
+                                        decoder_hidden_channels, out_channels)
+        self.epsilon = epsilon
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        self.representation_learner.reset_parameters()
+        self.balance_score_learner.reset_parameters()
+        self.treatment_decoder.reset_parameters()
+        self.control_decoder.reset_parameters()
+
+    def forward(self, tf: TensorFrame,
+                treatment_index: int) -> Tuple[Tensor, Tensor, Tensor]:
```

Comment:
Wouldn't this be cleaner if you don't merge `T` into `tf` in the first place? For example:
```python
def forward(self, tf, treatment_mask: Tensor)